woocommerce/woocommerce-ios
# [Store onboarding] Add `blog_public` to `Site`

Part of: #9477


## Description

Dashboard with "Launch your store" task
<img src="https://user-images.githubusercontent.com/524475/232741503-8d196d24-173e-4454-aa32-56a00ba849d2.png" width="320"/>

The "Launch your store" onboarding task is displayed as incomplete for stores that are already live. Many users have reported that this is confusing as their store has been launched a long time back. 

We inspected the problem, and it is an issue with the backend. Internal discussion -  p1681200289275869-slack-CNA89KY6M

While the issue is being fixed in the backend, we decided to release a workaround fix to hide the "Launch your store" task for already live sites. We are going to use the `blog_public` value from the `me/sites` response. Internal discussion - p1681726337528639-slack-C045CUK1Y3U


This PR,
- Adds `blog_public` property to `Site`. (Updates both Networking and Storage layers)
- Adds new core data model version and tests migration.
- Starts getting `blog_public` from `me/sites/` request from `AccountRemote`

## Testing instructions

1. Launch the app, and log out if needed.
2. Login back, and confirm that the store list is loaded correctly.
3. Select a store, and confirm it loads correctly.

## Screenshots

NA


---
